Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -460,6 +527,46 @@ public:
                                                     p_n_pairs_in_each_cell, pair_offsets.data());
             index_type* AMREX_RESTRICT p_pair_offsets = pair_offsets.dataPtr();
 
+            /*
+               preprocessing calculations for efficient gpu acceleration
+            */
+            amrex::Gpu::DeviceVector<index_type> n_ind_pairs_in_each_cell(n_cells+1);
+            index_type* AMREX_RESTRICT p_n_ind_pairs_in_each_cell = n_ind_pairs_in_each_cell.dataPtr();
+
+            amrex::ParallelFor( n_cells+1,
+                [=] AMREX_GPU_DEVICE (int i_cell) noexcept
+                {
+                    if (i_cell < n_cells)
+                    {
+                        const auto n_part_in_cell_1 = cell_offsets_1[i_cell+1] - cell_offsets_1[i_cell];
+                        const auto n_part_in_cell_2 = cell_offsets_2[i_cell+1] - cell_offsets_2[i_cell];
+                        // Particular case: no pair if a species has no particle in that cell
+                        if (n_part_in_cell_1 == 0 || n_part_in_cell_2 == 0)
```

Comment:
Is this `if` needed, since it would get the same result using the `min`?

Reply:
Agreed.